content/using_ec2_spot_instances_with_eks/scaling/test_hpa.md

```diff
@@ -0,0 +1,115 @@
+---
+title: "Scale the Application"
+date: 2018-08-07T08:30:11-07:00
+weight: 50
+---
+
+We are now ready to test the Horizontal Pod Autoscaler ! As a bonus we will also see the interaction between Cluster Autoscaler and Horizontal Pod Autoscaler.
+
+### Deploying the Stress CLI to Cloud 9
+
+To help us stress the application we will install a python helper app. The python helper application just calls in parallel on multiple process request to the monte-carlo-pi-service. This should generate load in our pods, which should also trigger the Horizontal Pod Autoscaler action for scaling the monte-carlo-pi-service replicaset.
+
+```
+mkdir -p ~/environment/submit_mc_pi_k8s_requests/
+curl -o ~/environment/submit_mc_pi_k8s_requests/submit_mc_pi_k8s_requests.py https://raw.githubusercontent.com/ruecarlo/eks-workshop-sample-api-service-go/master/stress_test_script/submit_mc_pi_k8s_requests.py
+chmod +x ~/environment/submit_mc_pi_k8s_requests/submit_mc_pi_k8s_requests.py
+curl -o ~/environment/submit_mc_pi_k8s_requests/requirements.txt https://raw.githubusercontent.com/ruecarlo/eks-workshop-sample-api-service-go/master/stress_test_script/requirements.txt
+sudo python3 -m pip install -r ~/environment/submit_mc_pi_k8s_requests/requirements.txt
+URL=$(kubectl get svc monte-carlo-pi-service | tail -n 1 | awk '{ print $4 }')
+~/environment/submit_mc_pi_k8s_requests/submit_mc_pi_k8s_requests.py -p 1 -r 1 -i 1 -u "http://${URL}"
+```
+
+The output of this command should show something like:
+```
+Total processes: 1
+Len of queue_of_urls: 1
+content of queue_of_urls: ab79391edde2d11e9874706fbc6bc60f-1090433505.eu-west-1.elb.amazonaws.com/?iterations=1
+100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 8905.10it/s]
+```
+
+### Scaling our Application and Cluster
+
+{{% notice note %}}
+Before starting the stress test, think what should be the expected outcome. Use **kube-ops-view** to verify that the changes you were expecting will
+happen over time. 
+{{% /notice %}}
+{{%expand "Show me how to get kube-ops-view url" %}}
+Execute the following command on Cloud9 terminal
+```
+kubectl get svc kube-ops-view | tail -n 1 | awk '{ print "Kube-ops-view URL = http://"$4 }'
+```
+{{% /expand %}}
+
+Run the stress test ! This time around we will run 2000 requests each expected to take ~1.3sec or so.
+```
+time ~/environment/submit_mc_pi_k8s_requests/submit_mc_pi_k8s_requests.py -p 100 -r 20 -i 30000000 -u "http://${URL}"
+```
+
+### Challenge 
+
+While the application is running, can you answer the following questions ?
```

Comment:
Remind people that they can open multiple terminals in Cloud9 to view different commands?

Reply:
Good point